src/Avalonia.Visuals/Rendering/SceneGraph/VisualNode.cs

```diff
@@ -193,9 +193,12 @@ public void SortChildren(Scene scene)
                 keys.Add(((long)zIndex << 32) + i);
             }
 
+            var toRemove = _children.ToList();
```

Comment:
It might be better to use a pooled list instance to reduce allocations.

Reply:
Yeah, but I figured that there's already so many allocations in the renderer that one more won't make much difference, and we can address when we address the rest of them.

Reply:
Allocating huge arrays (which List internally does)  can have a huge impact since they can end up in LOH.

Reply:
TBH it would require to have around 20k children to end up on LOH. Although we already have renderer objects in LOH (dirty visuals). We also reallocate all nodes, children, etc. per render. Maybe we should create new task for tracking potential renderer refactor.